projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DeploymentServiceV2.java

```diff
@@ -105,6 +107,27 @@ private void saveNewDesiredDeployment(DesiredDataJobDeployment deployment, DataJ
     desiredJobDeploymentRepository.save(deployment);
   }
 
+  /**
+   * Reads a data job deployment from the databse.
+   *
+   * @param dataJobName the job name
+   * @return empty optional or the job deployment
+   */
+  public Optional<DataJobDeploymentStatus> readDeploymentFromDB(String dataJobName) {
```

Comment:
THere is already public method public Optional<ActualDataJobDeployment> readDeployment. 

Seems unnecessary to have 2. 

Also ToModelApiConverter is mostly used in the Controller classes it seems not in the service classes.

Reply:
Antony is right, there is dedicate DeploymentModelConverter

Reply:
Also, the data job existence is already verified in the Controller.

Reply:
It may be better to reuse the `readDeployment` and do the conversion in the Controller.